### PR TITLE
E-717 feat: scopes refactor

### PR DESCRIPTION
**Pull request by dudiq**

### CODE REVIEW

**Comment by dudiq** [comment]
@lamartire ready for review

File: src/plugins/OauthPlugin/FrameStrategy/PopupFrame.js [hunk 1]
```diff
@@ -38,7 +38,12 @@ export default class PopupFrame extends BaseWindow {
       throw ConnectError.create(ERRORS.OAUTH_POPUP_CLOSED);
     }
     this.window.location.href = this.url;
-    this.window.resizeTo(DEFAULT_WIDTH, DEFAULT_HEIGHT);
+
+    try {
+      // Hack for cross origin access to call .resizeTo() method
+      this.window.resizeTo(DEFAULT_WIDTH, DEFAULT_HEIGHT);
+    } catch (e) {}
```
Comment (lamartire) [inline src/plugins/OauthPlugin/FrameStrategy/PopupFrame.js:45]: What will happen without `try / catch`?

File: src/plugins/OauthPlugin/Oauth/OauthPkceStrategy.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ export default class OauthPkceStrategy {
    * @param {string} oauthServer
    * @param {string} code
    * @param {StrategyParams} params
-   * @return {Promise<{expires: number, scope: string, token: string}>}
+   * @return {Promise<{expires: number, token: string}>}
```
Comment (lamartire) [inline src/plugins/OauthPlugin/Oauth/OauthPkceStrategy.js:73]: May be define types for responses too?

File: src/plugins/OauthPlugin/FrameStrategy/PopupFrame.js (hunk 1 above)
Comment (dudiq) [reply src/plugins/OauthPlugin/FrameStrategy/PopupFrame.js:45]: `.open` method will throw by cross domain error and flow will not work correctly

**Pull request merged**